src/test/java/pl/jablonskanycz/bakery/database/services/AddressServiceTest.java

```diff
@@ -0,0 +1,100 @@
+package pl.jablonskanycz.bakery.database.services;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import pl.jablonskanycz.bakery.database.domain.AddressEntity;
+import pl.jablonskanycz.bakery.database.repositories.AddressRepository;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest
+@ActiveProfiles("test")
+class AddressServiceTest {
+    @Autowired
+    private AddressService addressService;
+
+    @MockBean
+    private AddressRepository addressRepository;
+
+    @Test
+    public void shouldReturnAllAddresses() {
+        //given
+        AddressEntity address1 = AddressEntity.builder().addressId(1L).latitude(51.20).longitude(15.80).build();
+        AddressEntity address2 = AddressEntity.builder().addressId(2L).latitude(50.02).longitude(18.30).build();
+        when(addressRepository.findAll()).thenReturn(Arrays.asList(address1, address2));
+
+        //when
+        List<AddressEntity> addresses = addressService.getAllAddresses();
+
+        //then
+        assertNotNull(addresses);
+        assertEquals(2, addresses.size());
+    }
+
+    @Test
+    public void shouldAddAddress() {
+        //given
+        AddressEntity address = AddressEntity.builder().latitude(51.20).longitude(15.80).build();
+        when(addressRepository.save(any(AddressEntity.class))).thenReturn(address);
+
+        //when
+        addressService.addAddress(51.20, 15.80);
+
+        //then
+        verify(addressRepository).save(any(AddressEntity.class));
+    }
```

Comment:
in such a test it a good choice to test if a specific method was called. In more sophisticated tests we would probably ant to rather test the _outcome_ of the operation, not what was called underneath (not to test the inner implementation but to test the outcome).